src/core/common/random_manager.cpp

```diff
@@ -116,29 +119,106 @@ void RandomManager::NonCryptoPrng::Init(uint32_t aSeed)
         aSeed = 0x1;
     }
 
-    mState = aSeed;
+    mState = static_cast<uint32_t>(FLOAT_TO_FIXED(aSeed));
 }
 
-uint32_t RandomManager::NonCryptoPrng::GetNext(void)
+uint32_t RandomManager::NonCryptoPrng::FixedPointDivision(uint32_t aDividend, uint32_t aDivisor)
 {
-    uint32_t mlcg, p, q;
-    uint64_t tmpstate;
+    uint32_t divResult = static_cast<uint32_t>(((uint64_t)aDividend * (1 << FIXED_POINT_FRACTIONAL_BITS)) / aDivisor);
+    return divResult;
+}
 
-    tmpstate = (uint64_t)33614 * (uint64_t)mState;
-    q        = tmpstate & 0xffffffff;
-    q        = q >> 1;
-    p        = tmpstate >> 32;
-    mlcg     = p + q;
+uint32_t RandomManager::NonCryptoPrng::ComputeLcgRandom(uint32_t  aNumBits,
+                                                        uint32_t  aSeedVal,
+                                                        uint32_t *aDunif,
+                                                        uint32_t  aNdim)
+{
+    uint16_t i      = 0;
+    uint32_t DZ     = 0;
+    uint32_t DOVER  = 0;
+    uint32_t DZ1    = 0;
+    uint32_t DZ2    = 0;
+    uint32_t DOVER1 = 0;
+    uint32_t DOVER2 = 0;
+    uint32_t DTWO31 = 0;
+    uint32_t DMDLS  = 0;
+    uint32_t DA1    = 0;
+    uint32_t DA2    = 0;
+
+    DTWO31 = static_cast<uint32_t>(FLOAT_TO_FIXED(2147483648.0)); /* DTWO31=2**31  */
```

Comment:
This number is 1<<31 so after converting to fixed-point it won't fit in uint32_t.